File: source/static/js/modal.js

    $('.modal-trigger').click(function () {
        let cellId = $(this).data('cell-id');

        $.ajax({
            url: '/get_cell_data/',
            type: 'GET',
            data: {
                'cell_id': cellId
            },
            success: function (data) {
                $('#modal-lot').text(data.lot);
                $('#modal-company').text(data.company);
                $('#modal-name').text(data.name);
                $('#modal-price_per_unit').text(data.price_per_unit);
                $('#modal-quantity').text(data.quantity);
                $('#modal-planned_amount').text(data.planned_amount);
                $('#modal-delivery_deadline').text(data.delivery_deadline);
                $('#modal-proposed_product_name').text(data.proposed_product_name);
                $('#modal-supplier').text(data.supplier);
                $('#modal-price_without_discount').text(data.price_without_discount);
                $('#modal-supplier_discount').text(data.supplier_discount);
                $('#modal-price_with_discount').text(data.price_with_discount);
                $('#modal-vat').text(data.vat);
                $('#modal-note').text(data.note);
                $('#modal-manager').text(data.manager);
                $('#modal-purchase_price').text(data.purchase_price);
                $('#modal-date').text(data.date);
                $('#modal-deadline').text(data.deadline);
                $('#modal-procurement_type').text(data.procurement_type);
                $('#modal-paper_ad_link').text(data.paper_ad_link);
                $('#modal-lot_link').text(data.lot_link);
                $('#modal-profit_rate').text(data.profit_rate);
                $('#modal-delivery_rate').text(data.delivery_rate);
                $('#modal-purchase_price_per_unit').text(data.purchase_price_per_unit);
                $('#modal-bidding_price_per_unit').text(data.bidding_price_per_unit);
                $('#modal-budget_price_per_unit').text(data.budget_price_per_unit);
                $('#modal-overall_profit').text(data.overall_profit);
                $('#modal-overall_purchase_amount').text(data.overall_purchase_amount);
                $('#modal-overall_contract_amount').text(data.overall_contract_amount);
                $('#modal-winning_price').text(data.winning_price);
                $('#modal-commercial_offer_text').text(data.commercial_offer_text);
                $('#modal-status').text(data.status);
                $('#myModal').modal('show');
            },
            error: function (xhr, status, error) {
            }
        });
        $(document).on('click', function (e) {
            if ($(e.target).hasClass('modal')) {
                $('#myModal').modal('hide');
            }
        });

    });

File: source/smarttender/views/modal.py
from django.shortcuts import get_object_or_404
from django.http import JsonResponse
from smarttender.models import Tender


def get_cell_data(request):
    cell_id = request.GET.get('cell_id')
    tender = get_object_or_404(Tender, id=cell_id)
    data = {
        'lot': tender.lot,
        'company': tender.company,
        'name': tender.name,
        'price_per_unit': tender.price_per_unit,
        'quantity': tender.quantity,
        'planned_amount': tender.planned_amount,
        'delivery_deadline': tender.delivery_deadline,
        'proposed_product_name': tender.proposed_product_name,
        'supplier': tender.supplier,
        'price_without_discount': tender.price_without_discount,
        'supplier_discount': tender.supplier_discount,
        'price_with_discount': tender.price_with_discount,
        'vat': tender.vat,
        'note': tender.note,
        'manager': tender.manager,
        'purchase_price': tender.purchase_price,
        'overall_info': tender.overall_info,
        'date': tender.date,
        'deadline': tender.deadline,
        'procurement_type': tender.procurement_type,
        'paper_ad_link': tender.paper_ad_link,
        'lot_link': tender.lot_link,
        'profit_rate': tender.profit_rate,
        'delivery_rate': tender.delivery_rate,
        'purchase_price_per_unit': tender.purchase_price_per_unit,
        'bidding_price_per_unit': tender.bidding_price_per_unit,
        'budget_price_per_unit': tender.budget_price_per_unit,
        'overall_profit': tender.overall_profit,
        'overall_purchase_amount': tender.overall_purchase_amount,
        'overall_contract_amount': tender.overall_contract_amount,
        'winning_price': tender.winning_price,
        'commercial_offer_text': tender.commercial_offer_text,
        'status': tender.status,
    }
    return JsonResponse(data)












